File: rust_exercises/leetcode/Problem717/Solution.java
public class Solution {
    public boolean isOneBitCharacter(int[] bits) {
        int i = bits.length - 2;
        while (i >= 0 && bits[i] == 1) {
            i -= 1;
        }
        return (bits.length - i) % 2 == 0;
    }
}

File: rust_exercises/leetcode/Problem717/solution.rs
impl Solution {
    pub fn is_one_bit_character(bits: Vec<i32>) -> bool {
        let mut count: u32 = 0;
        for bit in bits.iter().rev().skip(1) {
            if *bit == 0 {
                break;
            }
            count += 1;
        }
        count % 2 == 0
    }
}
